### PR TITLE
8332899: RISC-V: add comment and make the code more readable (if possible) in MacroAssembler::movptr

### PR DESCRIPTION
**Pull request by Hamlin-Li**
Hi,
Can you help to review the patch?
As discussed, https://github.com/openjdk/jdk/pull/19246#discussion_r1613279908, it's worth to make the code more readable.
For movptr1, add some comments to help understand the tricky part.
For movptr2, it uses the similar (tricky) way as movptr1, so I align the code implementation with movptr1, and try to make it more straightforward.
I tried it, hope it's better.
Thanks.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8332899](https://bugs.openjdk.org/browse/JDK-8332899): RISC-V: add comment and make the code more readable (if possible) in MacroAssembler::movptr (**Enhancement** - P4)

### Reviewers
 * [Robbin Ehn](https://openjdk.org/census#rehn) (@robehn - **Reviewer**) ⚠️ Review applies to [93de6315](https://git.openjdk.org/jdk/pull/19431/files/93de63154b26e9fcbd8595a4ccbfcdbf9877ccf9)
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**) ⚠️ Review applies to [93de6315](https://git.openjdk.org/jdk/pull/19431/files/93de63154b26e9fcbd8595a4ccbfcdbf9877ccf9)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19431/head:pull/19431` \
`$ git checkout pull/19431`

Update a local copy of the PR: \
`$ git checkout pull/19431` \
`$ git pull https://git.openjdk.org/jdk.git pull/19431/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19431`

View PR using the GUI difftool: \
`$ git pr show -t 19431`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19431.diff">https://git.openjdk.org/jdk/pull/19431.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19431#issuecomment-2135570854)

### CODE REVIEW

**Review by robehn — APPROVED** [review]
Seems fine to me!

File: src/hotspot/cpu/riscv/macroAssembler_riscv.cpp [hunk 1]
```diff
@@ -1680,6 +1680,23 @@ void MacroAssembler::movptr(Register Rd, address addr, int32_t &offset, Register
 
 void MacroAssembler::movptr1(Register Rd, uint64_t imm64, int32_t &offset) {
   // Load upper 31 bits
+  //
+  // In case of 11th bit of `lower` is 0, it's straightforward to understand.
+  // In case of 11th bit of `lower` is 1, it's a bit tricky, to help understand,
+  // image divide both `upper` and `lower` into 2 parts respectively, i.e.
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/macroAssembler_riscv.cpp:1686]: Suggestion: s/image/imagine/

**Review by RealFYang — APPROVED** [review]
LGTM modulo the typo in code comment. Thanks.

File: src/hotspot/cpu/riscv/macroAssembler_riscv.cpp (hunk 1 above)
Comment (Hamlin-Li) [reply src/hotspot/cpu/riscv/macroAssembler_riscv.cpp]: Thanks, fixed.

**Comment by Hamlin-Li** [comment]
Thanks @robehn @RealFYang for your reviewing!

**Pull request closed without merging**